src/Controls/src/Core/Menu/MenuItem.cs

```diff
@@ -31,7 +31,12 @@ public partial class MenuItem : BaseMenuItem, IMenuItemController, IStyleSelecta
 		public static readonly BindableProperty IsDestructiveProperty = BindableProperty.Create(nameof(IsDestructive), typeof(bool), typeof(MenuItem), false);
 
 		/// <summary>Bindable property for <see cref="IconImageSource"/>.</summary>
-		public static readonly BindableProperty IconImageSourceProperty = BindableProperty.Create(nameof(IconImageSource), typeof(ImageSource), typeof(MenuItem), default(ImageSource));
+		public static readonly BindableProperty IconImageSourceProperty = BindableProperty.Create(nameof(IconImageSource), typeof(ImageSource), typeof(MenuItem), default(ImageSource),
+		 propertyChanged: (bindable, oldValue, newValue) => {
+				var menuItem = (MenuItem)bindable;
+				var icon = (ImageSource)newValue;
+				icon.Parent = menuItem;
+		 } );
```

Comment:
This seems to be the first time I have seen this. Are we doing this for other image sources?

Reply:
I don't think so

Reply:
Check for `null` on the new `ImageSource` before accessing the `.Parent` maybe? It is possible this could be a text only toolbar item.

Reply:
Parenting is one way of making sure the AppThemeBinding is propagated, ParentResourcesChanged another, and we probably don't need both

Reply:
I just re-tested and without the ParentResourcesChanged approach it works, but without parenting it doesn't 🤔